tsconfig.json

```diff
@@ -0,0 +1,10 @@
+{
+  "include": ["index.js"],
+  "compilerOptions": {
+    "allowJs": true,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "outDir": "dist",
+    "declarationMap": true
+  }
+}
```

Comment:
Let's improve the compiler options a bit.

Here is what we have in Desktop:

```json
  "compilerOptions": {
    "noEmit": true,
    "strict": true,
    "skipLibCheck": true,
    "allowJs": true,
    "checkJs": true,
    "lib": ["es2021"],
    "target": "es2021",
    "module": "commonjs",
    "moduleResolution": "node",

    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitReturns": true,
    "forceConsistentCasingInFileNames": true
  },
```

`skipLibCheck` is important to improve `tsc` speed.

`checkJs` is important to tell tsc to check our JS files and detect errors there too.

`lib` and `target` may be required to enable modern language features.

Personally, I tend to enable `strict`, because otherwise, it's easy to accidentally introduce `any` in the code, which then disables all type checks in the following code. The downside is that sometimes we need to describe types in more places, which is less ergonomic with JSDoc.

We need different `module` and `moduleResolution` values, since this is an ESM code base. I don't remember the correct values, though, TypeScript's support for ESM modules has been evolving quite a bit recently. See  https://www.typescriptlang.org/docs/handbook/esm-node.html

`forceConsistentCasingInFileNames` catches the problem when we use different letter casing in file names. macOS and Windows are not case-sensitive, but Linux is.

Reply:
I opened #115 to address this thread.